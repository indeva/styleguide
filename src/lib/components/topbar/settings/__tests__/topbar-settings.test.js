import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Settings from '..';

jest.useFakeTimers();

describe('Topbar Settings', () => {
  const defaultProps = {
    name: 'Felino de Santos',
    email: 'felino@teste.com',
    links: [
      { title: 'Editar perfil', icon: ['far', 'pen'], url: 'https://indeva.com.br/' },
      { title: 'Sair', icon: ['far', 'sign-out-alt'], url: 'https://indeva.com.br/beneficios/' },
    ],
    toggleSettings: () => {},
  };

  it('should renders properly', () => {
    const component = renderer.create(<Settings {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without active modifier', () => {
    const component = renderer.create(<Settings {...defaultProps} />);
    component.root.instance.setState({ visible: false });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onSettingsNavClick when clicking on nav links', () => {
    const onSettingsNavClick = jest.fn();
    const wrapper = shallow(<Settings {...defaultProps} onSettingsNavClick={onSettingsNavClick} />);
    wrapper.find('.topbar-settings__link').first().simulate('click');
    expect(onSettingsNavClick).toHaveBeenCalledWith({ title: 'Editar perfil', icon: ['far', 'pen'], url: 'https://indeva.com.br/' });
  });

  it('should remove class --active when clicking outsite', () => {
    const wrapper = mount(<Settings {...defaultProps} />);
    expect(wrapper.exists('.topbar-settings--active')).toEqual(true);
    wrapper.instance().handleClickOutside({ target: null });
    expect(wrapper.update().exists('.topbar-settings--active')).toEqual(false);
  });

  it('should call toggleSettings when clicking outsite', () => {
    const toggleSettings = jest.fn();
    const wrapper = mount(<Settings {...defaultProps} toggleSettings={toggleSettings} />);
    wrapper.instance().handleClickOutside({ target: null });
    jest.runAllTimers();
    expect(toggleSettings).toHaveBeenCalled();
  });
});
