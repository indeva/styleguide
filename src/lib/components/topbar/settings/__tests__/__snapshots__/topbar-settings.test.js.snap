// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Settings should render without active modifier 1`] = `
<div
  className="topbar-settings"
>
  <header
    className="topbar-settings__header"
  >
    Felino de Santos
    <span>
      felino@teste.com
    </span>
  </header>
  <nav
    className="topbar-settings__nav"
  >
    <a
      className="topbar-settings__link"
      href="https://indeva.com.br/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-pen fa-w-16 "
        data-icon="pen"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M493.26 56.26l-37.51-37.51C443.25 6.25 426.87 0 410.49 0s-32.76 6.25-45.25 18.74l-74.49 74.49L256 127.98 12.85 371.12.15 485.34C-1.45 499.72 9.88 512 23.95 512c.89 0 1.79-.05 2.69-.15l114.14-12.61L384.02 256l34.74-34.74 74.49-74.49c25-25 25-65.52.01-90.51zM118.75 453.39l-67.58 7.46 7.53-67.69 231.24-231.24 31.02-31.02 60.14 60.14-31.02 31.02-231.33 231.33zm340.56-340.57l-44.28 44.28-60.13-60.14 44.28-44.28c4.08-4.08 8.84-4.69 11.31-4.69s7.24.61 11.31 4.69l37.51 37.51c6.24 6.25 6.24 16.4 0 22.63z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Editar perfil
    </a>
    <a
      className="topbar-settings__link"
      href="https://indeva.com.br/beneficios/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-sign-out-alt fa-w-16 "
        data-icon="sign-out-alt"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M272 112v51.6h-96c-26.5 0-48 21.5-48 48v88.6c0 26.5 21.5 48 48 48h96v51.6c0 42.6 51.7 64.2 81.9 33.9l144-143.9c18.7-18.7 18.7-49.1 0-67.9l-144-144C323.8 48 272 69.3 272 112zm192 144L320 400v-99.7H176v-88.6h144V112l144 144zM96 64h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-53 0-96-43-96-96V160c0-53 43-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Sair
    </a>
  </nav>
</div>
`;

exports[`Topbar Settings should renders properly 1`] = `
<div
  className="topbar-settings topbar-settings--active"
>
  <header
    className="topbar-settings__header"
  >
    Felino de Santos
    <span>
      felino@teste.com
    </span>
  </header>
  <nav
    className="topbar-settings__nav"
  >
    <a
      className="topbar-settings__link"
      href="https://indeva.com.br/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-pen fa-w-16 "
        data-icon="pen"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M493.26 56.26l-37.51-37.51C443.25 6.25 426.87 0 410.49 0s-32.76 6.25-45.25 18.74l-74.49 74.49L256 127.98 12.85 371.12.15 485.34C-1.45 499.72 9.88 512 23.95 512c.89 0 1.79-.05 2.69-.15l114.14-12.61L384.02 256l34.74-34.74 74.49-74.49c25-25 25-65.52.01-90.51zM118.75 453.39l-67.58 7.46 7.53-67.69 231.24-231.24 31.02-31.02 60.14 60.14-31.02 31.02-231.33 231.33zm340.56-340.57l-44.28 44.28-60.13-60.14 44.28-44.28c4.08-4.08 8.84-4.69 11.31-4.69s7.24.61 11.31 4.69l37.51 37.51c6.24 6.25 6.24 16.4 0 22.63z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Editar perfil
    </a>
    <a
      className="topbar-settings__link"
      href="https://indeva.com.br/beneficios/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-sign-out-alt fa-w-16 "
        data-icon="sign-out-alt"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M272 112v51.6h-96c-26.5 0-48 21.5-48 48v88.6c0 26.5 21.5 48 48 48h96v51.6c0 42.6 51.7 64.2 81.9 33.9l144-143.9c18.7-18.7 18.7-49.1 0-67.9l-144-144C323.8 48 272 69.3 272 112zm192 144L320 400v-99.7H176v-88.6h144V112l144 144zM96 64h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-53 0-96-43-96-96V160c0-53 43-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Sair
    </a>
  </nav>
</div>
`;
